lib/friends/introvert.rb

```diff
@@ -462,18 +469,16 @@ def regex_friend_map
     #
     # The returned hash uses the following format:
     #   {
-    #     /regex/ => [list of locations matching regex]
+    #     location.name => [list of locations matching regex]
     #   }
     #
     # This hash is sorted (because Ruby's hashes are ordered) by decreasing
     # regex key length, so the key /Paris, France/ appears before /Paris/.
     #
-    # @return [Hash{Regexp => Array<Friends::Location>}]
+    # @return [Hash{String => Array<Regexp>}]
     def regex_location_map
-      @locations.each_with_object(Hash.new { |h, k| h[k] = [] }) do |location, hash|
-        location.regexes_for_name.each do |regex|
-          hash[regex] << location
-        end
+      @locations.each_with_object({}) do |location, hash|
```

Comment:
I'm so sorry—you changed this to exactly what I suggested but I made a typo—can we make this a hash of `{ regex => location }` instead to more closely match the `regex_friend_map`? That way, when we sort the keys by length, we know that `/Paris, France/` will be checked before `/Paris/` (which means we'd store activities as `Went shopping in _Paris, France_` instead of `Went shopping in _Paris_, France`).

Concretely I'm thinking something like:

```ruby
@locations.each_with_object({}) do |location, hash|
  location.regexes_for_name.each { |regex| hash[regex] = location }
end
```

Does that make sense?

Reply:
This makes sense to me. I change that, too